mm2src/coins/siacoin.rs

```diff
@@ -287,9 +290,36 @@ pub struct SiaFeeDetails {
 impl MmCoin for SiaCoin {
     fn spawner(&self) -> CoinFutSpawner { CoinFutSpawner::new(&self.abortable_system) }
 
-    fn get_raw_transaction(&self, _req: RawTransactionRequest) -> RawTransactionFut { unimplemented!() }
+    fn get_raw_transaction(&self, req: RawTransactionRequest) -> RawTransactionFut {
+        let tx_hash = match Hash256::from_str_no_prefix(&req.tx_hash) {
+            Ok(hash) => hash,
+            Err(e) => {
+                return Box::new(futures01::future::err(MmError::new(
+                    RawTransactionError::InvalidHashError(e.to_string()),
+                )))
+            },
+        };
+        self.get_tx_hex_by_hash(tx_hash.0.to_vec())
+    }
 
-    fn get_tx_hex_by_hash(&self, _tx_hash: Vec<u8>) -> RawTransactionFut { unimplemented!() }
+    fn get_tx_hex_by_hash(&self, tx_hash: Vec<u8>) -> RawTransactionFut {
+        let fut = async move {
+            let tx_req = GetEventRequest {
+                txid: Hash256::try_from(&tx_hash[..])
+                    .map_to_mm(|e| RawTransactionError::InvalidHashError(e.to_string()))?,
+            };
+            let _event = self
+                .client
+                .dispatcher(tx_req)
+                .await
+                .map_to_mm(|e| RawTransactionError::Transport(e.to_string()))?;
+            //Ok(RawTransactionRes { tx_hex: event }) // fixme: now what?
+            Err(RawTransactionError::InternalError(
+                "we have no hex serialization for siacoin?!".to_string(),
```

Comment:
No Sia app ever presents a transaction as a string of hex to a user. They are always presented as JSON strings.

If you need serialization for the purpose of sending over the wire, use serde_json::to_vec. This is a temporary solution until transaction decoding is finalized in sia-rust. 

If you need to serialize into hex to present to a user, it's a special case. We should aim to always provide the JSON string if it will be displayed to a user.